### DISCUSSION TITLE
How to get my device config

### DISCUSSION THREAD

**Discussion by netoxss** [amyasnikov/validity#101] [Q&A] [answered]
I add the device config path but it doesnt returns my device config, can anyone help me

![image](https://github.com/user-attachments/assets/179c065a-4940-4949-9510-6acf4a36bca5)

**Comment by amyasnikov** [comment]
Hey @netoxss, Device Config Path must contain the template of the path to the file(s) with device configuration.
Please consider the example from here:
https://validity.readthedocs.io/en/latest/entities/datasources/

If you still have no idea, I can help you with filling out this field, but I need more context. At least show me the **file structure** of your repository with device configurations.

**Reply by netoxss** [reply]
thats the file structure on gitlab, already defined the device path as .txt and nothing happens, always the same message: "config" is not a member of PORTARIA1 state.
<img width="431" alt="structure" src="https://github.com/user-attachments/assets/072b9554-b041-455b-bd63-545974ccbcb8">

**Reply by amyasnikov** [reply]
Is this the root of your repository on the screenshot? It's difficult to guess because it's cropped

And please show me the full page screenshot with the error

**Reply by netoxss** [reply]
error:
<img width="821" alt="errot" src="https://github.com/user-attachments/assets/f58bf6dd-42b2-4ac6-be9a-30ba58c524bd">
full structure:
<img width="815" alt="structure" src="https://github.com/user-attachments/assets/2e37c10d-a5e4-4d20-88b7-df70f5e9de06">
Data source config:
<img width="374" alt="config" src="https://github.com/user-attachments/assets/856f9eb3-1093-43c4-ab8c-39a9675f97ef">

**Reply by netoxss** [reply]
url:
![image](https://github.com/user-attachments/assets/78bdce0d-7665-4aa3-acc9-671ffcbed19f)

**Reply by amyasnikov** [reply, marked as answer]
1. The files in your repo do not contain ".txt", so you don't need it in Device Config Path as well. Your Device Config Path has to be just `{{ device.name }}`.
2. If you fill Device Config Path in the right way, The Metainfo block on your screen must contain the link to the Data File (there is a dash instead on your screen). If this does not happen, make sure your Data Source is synchronized with your repo and contains the Data File with the config.
3. Make sure you have Serializer bound to the Device (it can be done at Manufacturer, Device Type or individual Device levels according to your needs).

**Reply by netoxss** [reply]
thx, now its working, i have to set as {{ device.name }} and not {{device.name}}, and too selected in the device a config serializer
